packages/firestore/src/local/target_data.ts

```diff
@@ -78,17 +83,19 @@ export class TargetData {
       sequenceNumber,
       this.snapshotVersion,
       this.lastLimboFreeSnapshotVersion,
-      this.resumeToken
+      this.resumeToken,
+      this.expectedCount
     );
   }
 
   /**
-   * Creates a new target data instance with an updated resume token and
-   * snapshot version.
+   * Creates a new target data instance with an updated resume token,
+   * snapshot version, and expectedCount if presented.
    */
   withResumeToken(
     resumeToken: ByteString,
-    snapshotVersion: SnapshotVersion
+    snapshotVersion: SnapshotVersion,
+    expectedCount?: number
```

Comment:
I don't really like having `expectedCount` be optional here. It makes it too easy to forget to specify it. Is it difficult to drop the `?` and make it a required argument?